src/highlight.js

```diff
@@ -70,7 +70,7 @@ const HLJS = function(hljs) {
     ignoreUnescapedHTML: false,
     throwUnescapedHTML: false,
     noHighlightRe: /^(no-?highlight)$/i,
-    languageDetectRe: /\blang(?:uage)?-([\w-]+)\b/i,
+    languageDetectRe: /\blang(?:uage)?-(\w[\w+-]*)/i,
```

Comment:
I think we need to replace the `\b` with something to avoid partial matches of a longer string... perhaps something like `(?:$|(?=\s))` ie end of string or a trailing space... since we execute this against the entire css class name string...